quickstart/101-batch-account-with-storage/main.tf

```diff
@@ -0,0 +1,40 @@
+resource "random_pet" "rg_name" {
+  prefix = var.resource_group_name_prefix
+}
+
+resource "azurerm_resource_group" "rg" {
+  name     = random_pet.rg_name.id
+  location = var.resource_group_location
+}
+
+resource "random_string" "azurerm_storage_account_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "random_string" "azurerm_batch_account_name" {
+  length  = 13
+  lower   = true
+  numeric = false
+  special = false
+  upper   = false
+}
+
+resource "azurerm_storage_account" "storage" {
+  name                     = "storage${random_string.azurerm_storage_account_name.result}"
+  resource_group_name      = azurerm_resource_group.rg.name
+  location                 = azurerm_resource_group.rg.location
+  account_tier             = element(split("_", var.storage_account_type), 0)
```

Comment:
If you are going to do this string manipulation, you need to make sure to have validation in the variable to ensure that things don't go sideways when someone enters a string that cannot be split this way.

Reply:
Thanks, @grayzu. I've fixed it.